src/components/molecules/TableComponent/TableComponent.tsx

```diff
@@ -15,55 +33,149 @@ const TableComponent: React.FunctionComponent<TableComponentPropsType> = ({
   table,
   tableLocked,
   venue,
+  type,
 }) => {
   const { openUserProfileModal } = useProfileModalControls();
   const locked = tableLocked(table.reference);
   const numberOfSeatsLeft = table.capacity && table.capacity - users.length;
   const full = numberOfSeatsLeft === 0;
-  return (
-    <div className={`table-component-container ${table.reference}`}>
-      <div
-        className="table-item"
-        style={{
+  const { userId } = useUser();
+
+  const { isAdminUser, isLoading: isCheckingRole } = useIsAdminUser(userId);
+
+  const isRemoveButtonShown = !isCheckingRole && isAdminUser;
+
+  const {
+    isShown: isModalShown,
+    toggle: toggleModal,
+    hide: hideModal,
+  } = useShowHide(false);
+
+  const [{ loading: isDeletingTable }, removeTable] = useAsyncFn(async () => {
+    if (!venue.id) return;
+
+    await deleteTable({
+      venueId: venue.id,
+      tableName: table.reference,
+      defaultTables:
+        venue.template === VenueTemplate.jazzbar
+          ? JAZZBAR_TABLES
+          : CONVERSATION_TABLES,
+    });
+
+    toggleModal();
+  }, [table.reference, venue.id, venue.template, toggleModal]);
+
+  const isDeleteButtonDisabled = isDeletingTable;
```

Comment:
No gain of a simple alias, might as well just use `isDeletingTable` for the button prop
```suggestion
```

Reply:
done 2ba4481